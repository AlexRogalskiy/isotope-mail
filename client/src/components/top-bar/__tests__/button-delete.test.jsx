import React from 'react';
import {shallow} from 'enzyme/build/index';
import ButtonDelete from '../button-delete';

describe('ButtonDelete component test suite', () => {
  test('Snapshot render, should render button', () => {
    // Given
    const props = {onClick: jest.fn()};
    // When
    const button = shallow(<ButtonDelete {...props}/>);
    // Then
    expect(button).toMatchSnapshot();
  });
  test('click, should trigger function', () => {
    // Given
    const props = {onClick: jest.fn()};
    const button = shallow(<ButtonDelete {...props}/>);
    // When
    button.find('button').simulate('click');
    // Then
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
