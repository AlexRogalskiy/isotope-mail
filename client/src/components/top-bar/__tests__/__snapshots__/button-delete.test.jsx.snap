// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDelete component test suite Snapshot render, should render button 1`] = `
<button
  className="material-icons mdc-top-app-bar__action-item"
  onClick={[MockFunction]}
>
  delete
</button>
`;
